src/mpid/ch4/subconfigure.m4

```diff
@@ -333,13 +333,23 @@ fi
 
 AC_ARG_WITH(ch4-max-vcis,
     [--with-ch4-max-vcis=<N>
-       Select max number of VCIs to configure (default is 1; minimum is 1)],
-    [], [with_ch4_max_vcis=1 ])
-if test $with_ch4_max_vcis -le 0 ; then
-   AC_MSG_ERROR(Number of VCIs must be greater than 0)
+       Select max number of VCIs to configure (default is 1; minimum is 1; maximum is 64)],
+    [], [with_ch4_max_vcis=default])
+
+if test "$with_ch4_max_vcis" = "default" ; then
+    if test $thread_granularity = MPICH_THREAD_GRANULARITY__VCI ; then
+        with_ch4_max_vcis=64
+    else
+        with_ch4_max_vcis=1
+    fi
+else
+    if test $thread_granularity != MPICH_THREAD_GRANULARITY__VCI ; then
+        AC_MSG_ERROR(Option --with-ch4-max-vcis requires --enable-thread-cs=per-vci)
+    fi
 fi
-if test $with_ch4_max_vcis -gt 1 -a $thread_granularity != MPICH_THREAD_GRANULARITY__VCI ; then
-    AC_MSG_ERROR(CH4_MAX_VCIS greater than 1 requires --enable-thread-cs=per-vci)
+
+if test $with_ch4_max_vcis -lt 1 -o $with_ch4_max_vcis -gt 64; then
+   AC_MSG_ERROR(Number of VCIs must be between 1 and 64)
```

Comment:
Did you think of any specific reason to cap the number of vcis at 64?

Reply:
Yes, the request handle currently use 6 bits to code pools, thus capped at 64.

Reply:
I see. I think it would be good to document it somewhere (if not done already). Maybe as a code comment here?